### PR TITLE
SwiftUI contact list

### PR DESCRIPTION
**Pull request by matthewrfennell**
# Introduction

This PR implements the first part of #1094

To do in (if it's ok) future PRs:

- Use `ObservableKVOWrapper<MLContact>` everywhere for live updates
- Implement notification badge for `ContactEntry` to match the old contacts screen
- Delete unused code from `ContactsViewController.m` etc.
- #121

# Videos

## iPhone

https://github.com/user-attachments/assets/60c4f99f-fb5d-4bab-ba37-6a3662573d6e

## iPad

At 2:22, the contact gets deleted from another device.

https://github.com/user-attachments/assets/990247f4-475e-41a4-99ad-f22e6f260401

## Animated notification badge

Just before raising this PR, I noticed the contact request notification badge wasn't animated.
I didn't want to re-record everything, so here's a recording of just that animation.

https://github.com/user-attachments/assets/5bafecbe-fab8-4e68-87a1-d9216fce8ea8

### CODE REVIEW

File: Monal/Classes/ContactsView.swift [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+                                .tint(.blue)
+                                .imageScale(.large)
+                        }
+                        .accessibilityLabel("Open contact details")
+                    }
+                }
+            }
+        }
+        .swipeActions(allowsFullSwipe: false) {
+            // We do not use a Button with destructive role here as we would like to display the confirmation dialog first.
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/ContactsView.swift:71]: That's very nice! could you add that behaviour to the group members list, too?

File: Monal/Classes/ContactsView.swift [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+            }
+        }
+        .sheet(isPresented: $createGroupIsPresented) {
+            NavigationStack {
+                CreateGroupMenu(delegate: SheetDismisserProtocol())
+                    .modifier(Dismissable(delegate: delegate))
+            }
+        }
+        .sheet(isPresented: $addContactIsPresented) {
+            NavigationStack {
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/ContactsView.swift]: Would be nicer to not open a new navigation stack here, but use the current one (that would look nice even on ipad/macos, I guess) and let the user return to the contact list by using the back button.

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]

**Comment by matthewrfennell** [comment]
# Popover

Here's how it looks now:

https://github.com/user-attachments/assets/d2aa2eff-910a-41ba-8767-ef528ab720a5

There is one edge case: if the view size changes from regular to compact (or vice versa) while the view is open, the button doesn't get added or removed:

https://github.com/user-attachments/assets/dee7262e-f553-42d2-9b7c-7aac94b39d6c

I'm looking into this now, but it might take some time to work out. It may be easier to wait until the ActiveChatsViewController is also SwiftUI.

# Group list swipe action

Swipe action now doesn't allow full swipe, and users joining and leaving are now animated (presence changes are not animated yet).

https://github.com/user-attachments/assets/f1b3ab9c-36d7-4b7e-95f3-3fb93405b974

Here's how the sheet looks on the iPhone:

group-swipe-action

(sheet looks the same as before in both cases, just wanted to show it)

**Comment by tmolitor-stud-tu** [comment]
Regarding the back button stuff: have a look at `MLSplitViewDelegate.m`.

This should allow you to trigger an event if the split view type (compact/expanded) changes and react accordingly in swiftui.

It will allow you to remove the code passing in the state (`horizontalSizeClass`, `shouldDisplayBackButton` etc.) all the way from active chats down to the swiftui view, too :)

**Comment by matthewrfennell** [comment]
GitHub is getting stuck uploading the video. So here's a blurry gif instead :)

https://github.com/user-attachments/assets/9af4f125-b06f-4377-a179-231c8479e18b

**Comment by tmolitor-stud-tu** [comment]
I feel like `shouldAlwaysDisplayBackButton`isn't needed anymore, right?

Is the `SizeClassWrapper` needed? Wouldn't it be better to just add that `horizontal` property to `ActiveChats` directly (renamed to `horizontalSizeClass`, of course)

**Comment by matthewrfennell** [comment]
Here's how the `ContactDetails` view dismiss button looks now:

https://github.com/user-attachments/assets/6c53de4a-93bd-4b17-b6bd-4e048652fc53

File: Monal/Classes/ContactsView.swift (hunk 1 above)
Comment (tmolitor-stud-tu) [reply Monal/Classes/ContactsView.swift:71]: Don't forget this one :)

**Pull request merged**